src/modelsGen.ts

```diff
@@ -168,6 +179,17 @@ async function main() {
   template += `export interface Models {\n${modelsExportParts
     .map((name) => `${name}: ${name}`)
     .join(';\n')}\n}\n\n`;
+
+  // openApi spec doesn't support a boolean that always false or always true
+  // however union types with forced booleans are very valuable for type
+  // narrowing in typescript. The following regex replace might be brittle,
+  // but if it doesn't find these cases, the types will just be a litte worse.
+  // Note this relies on the addCommentInfo function to add the "Always false"
+  // comments from spec descriptions in the typescript, if these descriptions
+  // change in the spec, this will need to be updated
+  template = template.replaceAll(/boolean.+\/\* Always false \*\//g, 'false');
+  template = template.replaceAll(/boolean.+\/\* Always true \*\//g, 'true');
```

Comment:
CC @jessfraz @adamchalmers 
I was unsure if this was a good idea since it's bending the rules and going outside of OpenAPI, but I think it does result in better TS types, example:
![image](https://github.com/KittyCAD/kittycad.ts/assets/29681384/5acf17ea-de8f-439f-927a-3b8c542b5b45)